### PR TITLE
gh-94808: add tests covering `PySequence_{Set,Del}Slice`

### PR DESCRIPTION
**Pull request by sobolevn**
These two method were not covered:
Снимок экрана 2022-11-05 в 15 12 39
Снимок экрана 2022-11-05 в 15 12 29

I haven't covered `null_error();` case, because it is impossible to reach from python code.
And it is also quite trivial not to bother writing its own C-level tests.

* Issue: gh-94808

### CODE REVIEW

**Review by iritkatriel — APPROVED** [review]

File: Lib/test/test_capi.py [hunk 1]
```diff
@@ -416,6 +416,70 @@ class SubDict(dict):
...
+            def __setitem__(self, index, value):
+                self.index = index
+                self.value = value
+
+        c = Custom()
+        _testcapi.sequence_set_slice(c, 0, 5, 'abc')
+        self.assertEqual(c.index, slice(0, 5))
+        self.assertEqual(c.value, 'abc')
+
+        # Immutable sequences must raise:
```
Comment (iritkatriel) [inline Lib/test/test_capi.py:440]: Here it would be good to check that failed slices didn't change the sequence.

Comment (sobolevn) [reply Lib/test/test_capi.py:440]: Good idea! Done.

Thank you for all the reviews 🎉

File: Lib/test/test_capi.py [hunk 2]
```diff
@@ -437,17 +437,24 @@ def __setitem__(self, index, value):
...
-        with self.assertRaises(TypeError):
-            _testcapi.sequence_set_slice({1: 'a', 2: 'b', 3: 'c'}, 1, 3, 'xy')
+        # Wrong cases:
+        import copy
+
+        bad_calls = [
+            # Immutable sequences must raise:
+            ((1, 2, 3, 4), 1, 3, (8, 9)),
+            ('abcd', 1, 3, 'xy'),
+            # Not a sequnce:
```
Comment (iritkatriel) [inline Lib/test/test_capi.py]: typo: sequnce

File: Lib/test/test_capi.py [hunk 3]
```diff
@@ -437,17 +437,24 @@ def __setitem__(self, index, value):
...
+            (None, 1, 3, 'xy'),
+            ({1: 'a', 2: 'b', 3: 'c'}, 1, 3, 'xy'),  # is a mapping
+        ]
+
+        for seq, i1, i2, obj in bad_calls:
+            with self.subTest(seq=seq, i1=i1, i2=i2, obj=obj):
+                seq_copy = copy.copy(seq)
```
Comment (iritkatriel) [inline Lib/test/test_capi.py]: for tuples and strings copy returns the same object:

```
>>> s = "blah"
>>> s is copy.copy(s)
True
>>> s = (1,2,3)
>>> s is copy.copy(s)
True
```

**Review by iritkatriel — APPROVED** [review]

**Comment by sobolevn** [comment]
I've seen the buildbot failure. Going to monitor other builds: if they have the same problem, I will fix it.

Thanks for the merge!

**Pull request merged**